### PR TITLE
Store terraform ROSA provider build on every merge

### PR DESCRIPTION
**Pull request by enriquebelarte**
Build image with every change in terraform-redhat-provider-ocm.
https://issues.redhat.com/browse/OCM-827

### CODE REVIEW

File: ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-main__images.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+binary_build_commands: make install
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.18
+images:
+- dockerfile_path: build/ci-tf-e2e.Dockerfile
```
Comment (osherdp) [inline ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-main__images.yaml:8]: isn't it already being built [here](https://github.com/openshift/release/blob/f7ac3ee42a5bdb79becfb118ac5c8baa98e8acbb/ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-main.yaml#L8)?
I think we just need to add the ``promotion`` stanza there

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-main__images.yaml (hunk 1 above)
Comment (enriquebelarte) [reply ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-main__images.yaml:8]: You are right. Adding it to the existing one. Thank you.

**Review by osherdp — APPROVED** [review]

**Review by bardielle — APPROVED** [review]
/lgtm

**Comment by enriquebelarte** [comment]
pj-rehearse ack

**Pull request merged**